### PR TITLE
[PTQ][ONNX][OV] Quantized embedding weights

### PR DESCRIPTION
**Pull request by KodiaqQ**
### Changes

- Added embedding weights quantization

### Reason for changes

- Proper quantization scheme

### Related tickets

- 113692

### Tests

- Added graph for `tests/torch/ptq/test_graphs.py`
- Added graph for `tests/onnx/quantization/test_graphs.py`
- Updated graph for `tests/openvino/native/quantization/test_graphs.py`

Old (no-) FQ placement (OV):
image

New FQ placement (OV):
image

New FQ placement (ONNX):
image

### CODE REVIEW

**Comment by KodiaqQ** [comment]
run pylint pre-commit tests
run pytorch pre-commit tests
run tensorflow pre-commit tests
run openvino pre-commit tests
run onnx pre-commit tests

File: nncf/openvino/quantization/default_quantization.py [hunk 1]
```diff
@@ -53,5 +53,5 @@
     ],
     QuantizationTrait.QUANTIZATION_AGNOSTIC: QUANTIZE_AGNOSTIC_OPERATIONS,
     QuantizationTrait.CONCAT: [ov_metatypes.OVConcatMetatype],
-    QuantizationTrait.OUTPUT_QUANTIZATION_AS_WEIGHTS: [],
+    QuantizationTrait.OUTPUT_QUANTIZATION_AS_WEIGHTS: [ov_metatypes.OVEmbeddingMetatype],
```
Comment (kshpv) [inline nncf/openvino/quantization/default_quantization.py:56]: Should it be also added for ONNX?

File: nncf/onnx/graph/metatypes/onnx_metatypes.py [hunk 2]
```diff
@@ -709,3 +765,23 @@ def _is_depthwise_conv(model: onnx.ModelProto, node: onnx.NodeProto) -> bool:
...
+    """
+    Returns True if the layer can be represented as embedding, False - otherwise.
+
+    :param model: ONNX model to get the node's weight.
+    :param node: Layer to check whether it is embedding.
+    :return: True if the layer is embedding, False - otherwise.
+    """
+    tensor_port_id = ONNXEmbeddingMetatype.weight_port_ids[0]
+    onnx_graph = ONNXGraph(model)
+    allowed_types_list = ["TensorProto.FLOAT", "TensorProto.DOUBLE"]
```
Comment (kshpv) [inline nncf/onnx/graph/metatypes/onnx_metatypes.py:780]: TensorProto.DOUBLE also is not supported for quantization. Take a look at https://github.com/openvinotoolkit/nncf/blob/develop/nncf/onnx/graph/nncf_graph_builder.py#L327
So, I recommend removing it from the list for now and putting a comment or creating a ticket.

**Review by kshpv — CHANGES_REQUESTED** [review]

**Comment by KodiaqQ** [comment]
run tensorflow pre-commit tests

**Comment by KodiaqQ** [comment]
run pytorch pre-commit tests
run tensorflow pre-commit tests

**Comment by KodiaqQ** [comment]
@l-bat, @vshampor, @kshpv, @andrey-churkin, review, please.

**Review by l-bat — APPROVED** [review]

File: nncf/openvino/graph/metatypes/openvino_metatypes.py [hunk 3]
```diff
@@ -403,6 +404,16 @@ class OVLogicalXorMetatype(OVOpMetatype):
     hw_config_names = [HWConfigOpName.LOGICALXOR]
 
 
+@OV_OPERATOR_METATYPES.register()
+class OVEmbeddingMetatype(OVOpMetatype):
```
Comment (l-bat) [inline nncf/openvino/graph/metatypes/openvino_metatypes.py:408]: Why does `ONNXEmbeddingMetatype` have`weight_port_ids` but `const_channel_axis` does not exist for `OVEmbeddingMetatype`?

**Review by andrey-churkin — APPROVED** [review]
LGTM

File: nncf/openvino/graph/metatypes/openvino_metatypes.py (hunk 3 above)
Comment (KodiaqQ) [reply nncf/openvino/graph/metatypes/openvino_metatypes.py:408]: For now, there is no configuration for weights in `cpu.json` for the `Embedding` type. This is why the `const_channel_axis` wouldn't have any effect. To be honest, `weight_port_ids` wouldn't have any effect too.
Should we remove or extend the current version, what you do think?

**Review by kshpv — APPROVED** [review]
👍

**Review by l-bat — APPROVED** [review]

**Review by vshampor — APPROVED** [review]

**Comment by KodiaqQ** [comment]
run pytorch pre-commit tests

**Pull request merged**